src/index.ts

```diff
@@ -273,16 +276,11 @@ export const plugin: Plugin<[FlexibleContainerOptions?], Root> = (options) => {
    * checks the paragraph node starts with a Text Node;
    * and checks the value starts with container start marker.
    */
-  function checkIsTarget(node: Paragraph): boolean {
+  function getOpeningFenceLength(node: Paragraph): number {
```

Comment:
I suppose it is more feasible to return fence (string) or undefined  (namely getOpeningFence)
```ts
  function getOpeningFence(node: Paragraph): string | undefined {
    const firstElement = node.children[0];

    if (firstElement.type !== "text") return;

    const match = firstElement.value.match(/^:{3,}/u);
    return match ? match[0] : undefined;
  }
```